runtime-modules/content/src/lib.rs

```diff
@@ -575,94 +655,89 @@ decl_module! {
 
-            // channel has a dynamic bag associated to it -> remove assets from storage
-            if let Ok(bag) = T::DataObjectStorage::ensure_bag_exists(&bag_id) {
-                // ensure that bag size provided is valid
-                ensure!(
-                    bag.objects_number == num_objects_to_delete,
-                    Error::<T>::InvalidBagSizeSpecified
-                );
+            //
+            // == MUTATION SAFE ==
+            //
+            Self::execute_delete_channel_mutation(sender, channel_id, if bag_exists { Some(&dynamic_bag_id) } else { None }, &assets_to_remove)?;
 
-                // construct collection of assets to be removed
-                let assets_to_remove = T::DataObjectStorage::get_data_objects_id(&bag_id);
 
-                if !assets_to_remove.is_empty() {
-                    Storage::<T>::can_delete_dynamic_bag_with_objects(
-                        &dyn_bag,
-                    )?;
+            // deposit event
+            Self::deposit_event(RawEvent::ChannelDeleted(actor, channel_id));
 
-                    Storage::<T>::can_delete_data_objects(
-                        &bag_id,
-                        &assets_to_remove,
-                    )?;
-                } else {
-                    Storage::<T>::can_delete_dynamic_bag(
-                        &dyn_bag,
-                    )?;
-                }
+            Ok(())
+        }
 
-                //
-                // == MUTATION SAFE ==
-                //
+        // extrinsics for channel deletion as moderator
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn delete_channel_as_moderator(
+            origin,
+            actor: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
+            channel_id: T::ChannelId,
+            num_objects_to_delete: u64,
+            rationale: Vec<u8>,
+        ) -> DispatchResult {
 
-                // remove specified assets from storage
-                if !assets_to_remove.is_empty() {
-                    Storage::<T>::delete_data_objects(
-                        sender.clone(),
-                        Self::bag_id_for_channel(&channel_id),
-                        assets_to_remove.clone(),
-                    )?;
-                }
+            let sender = ensure_signed(origin)?;
+            // check that channel exists
+            let channel = Self::ensure_channel_validity(&channel_id)?;
 
-                // delete channel dynamic bag
-                Storage::<T>::delete_dynamic_bag(
-                    sender,
-                    dyn_bag,
-                )?;
-            }
+            // Permissions check
+            let actions_to_perform = if num_objects_to_delete == 0 {
+                vec![ContentModerationAction::DeleteChannel]
+            } else {
+                vec![ContentModerationAction::DeleteChannel, ContentModerationAction::DeleteObject]
```

Comment:
I removed the distinction between removing a video with assets and a video without assets, but I still think we need separate extrinsics to remove channel/video assets by curators, because now that removing a video which has an nft issued is impossible, this opens up an attack vector where one can publish sensitive content, issue nft for it and therefore prevent the sensitive video asset from beeing removed (because it can only be removed along with the video, which in that case would not possible due to an nft beeing issued)